### DISCUSSION TITLE
Saving velocities for all particles

### DISCUSSION THREAD

**Discussion by TheDoctorWho169** [glotzerlab/hoomd-blue#1715] [Support] [answered]
Hello, 
I would like to save the velocities of all particles in a gsd file ever n timesteps for some post processing work. Below is my current attempt at saving the velocities which is based on https://hoomd-blue.readthedocs.io/en/v4.4.1/module-hoomd-write.html#hoomd.write.GSD. 

    gsd_velocity_writer = hoomd.write.GSD(filename='velocity.gsd',
                                trigger=hoomd.trigger.Periodic(1000),
                                mode='ab',
                                dynamic=['momentum'])
    sim.operations.writers.append(gsd_velocity_writer)

The problem is that when I try to read the file using the code below.

def velocity():
    rawvel = gsd.hoomd.open('velocity.gsd', mode='r')
    print(rawvel[0].log.keys())

The terminal outputs:

dict_keys([ ])

Am I writing/reading the file incorrectly? Is there a better way to save the velocities of the particles?

Thank you!

**Comment by joaander** [comment, marked as answer]
Use `rawvel[0].particles.velocity`. This is documented at https://gsd.readthedocs.io/en/v3.2.1/python-module-gsd.hoomd.html#gsd.hoomd.ParticleData.velocity

**Reply by TheDoctorWho169** [reply]
Thank you very much!
